index.js

```diff
@@ -228,11 +228,17 @@ class ServerlessCustomDomain {
         hZoneName = hZoneName.substr(0, hostedZone.Name.length - 1);   // Takes out the . at the end
         return this.givenDomainName.includes(hZoneName);
       });
-      hostedZoneId = hostedZoneId.Id;
-      // Extracts the hostzone Id
-      const startPos = hostedZoneId.indexOf('e/') + 2;
-      const endPos = hostedZoneId.length;
-      return hostedZoneId.substring(startPos, endPos);
+      if (hostedZoneId) {
+        hostedZoneId = hostedZoneId.Id;
+        // Extracts the hostzone Id
+        const startPos = hostedZoneId.indexOf('e/') + 2;
+        const endPos = hostedZoneId.length;
+        return hostedZoneId.substring(startPos, endPos);
+      }
+      return null;
```

Comment:
I think its better to throw an exception instead of returning null if its unable to find the hosted zone. With the `createRoute53Record` argument, this code will only be called if a hosted zone is expected to exist so its an error if it doesn't.